core/src/main/scala/pink/cozydev/protosearch/internal/NextPositionToMatch.scala

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2022 CozyDev
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pink.cozydev.protosearch.internal
+
+object NextPositionToMatch {
+
+  def indexOfMax(positions: Array[Int]): Int = {
+    var i, maxI = 0
+    while (i < positions.size) {
+      maxI = if (positions(i) > positions(maxI)) i else maxI
+      i += 1
+    }
+    maxI
+  }
+
+  def nextNotInOrderWithLargest(positions: Array[Int]): Int = {
```

Comment:
not sure if you wanted to do the rename to `nextNotInToleranceWithLargest` now, or if you want to wait and do the rename when adding the slop param? 

I have no preference on when the rename happens

Reply:
I'm leaning towards waiting until the "slop" support, which I hope is soon, because it seems cool and weirdly niche.